fedimint-client/src/db.rs

```diff
@@ -38,6 +38,26 @@ pub enum DbKeyPrefix {
     ClientInviteCode = 0x30,
     ClientInitState = 0x31,
     ClientMetadata = 0x32,
+    /// Arbitrary data of the applications integrating Fedimint client and
+    /// wanting to store some Federation-specific data in Fedimint client
+    /// database.
+    ///
+    /// New users are encouraged to use this single prefix only.
+    //
+    // TODO: https://github.com/fedimint/fedimint/issues/4444
+    //       in the future, we should make all global access to the db private
+    //       and only expose a getter returning isolated database.
+    UserData = 0xb0,
+    /// Prefixes between 0xb1..=0xcf shall all be considered allocated for
+    /// historical and future external use
+    ExternalReservedStart = 0xb1,
+    /// Prefixes between 0xb1..=0xcf shall all be considered allocated for
+    /// historical and future external use
+    ExternalReservedEnd = 0xcf,
+    /// 0xd0.. reserved for Fedimint internal use
+    InternalReservedStart = 0xd0,
```

Comment:
Why would we have an internal reserve? I'd say unless something is marked as "External" integrators don't have any business writing in our DB.

Reply:
Yeah. It's not necessary, but let's book it so even we are not adding anything there so we have a block of contiguous prefixes for special purposes in the future etc.

I basically started by booking the whole range up to 0xff for users, but 0xff is already used for special stuff (modules) so did it this way.